### PR TITLE
added support of the tokens.json, set all the tab controls to be scrollable #2282

### LINKED ISSUE #2282: Improve Segemented Controls and Token Groups
This is corresponding to https://github.com/AlphaWallet/alpha-wallet-ios/issues/3546

Another issue related to the Wallet tab. We are very, very close to the end design I made while designing Samoa.
So based on Hu's recent work, we now have specified token groups.

- All (Assets, Defi, Governance but no Collectibles)
- Assets (ERC20)
- Collectibles (ERC721 and ERC1155),
- DeFi (all ctokens, atokens, ptokens, itokens, liquidity pools, etc...),
- Governance (DAO tokens like COMP, UNI, etc)

So we have to:
1. Update Scrollable Tabs (so they are horizontally scrollable) https://material.io/components/tabs
2. Add Token groups

**Zeplin:**
https://zpl.io/2ZRJvlE

**Prototype**
https://invis.io/6U125DXFWE2X

Wallet-Long

### PR DESCRIPTION
**Pull request by mpaschenko**
- weight function (order) of the groups should be implemented, atm just hash (TokenSortGroup)
- tokens.json has to be added to the tokens repo, and URL has to be updated (TokensMappingRepository)

Closes #2282

### CODE REVIEW

File: app/src/main/java/com/alphawallet/app/entity/TokensMapping.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.List;
+
+public class TokensMapping {
+
+    public class Contract {
+
+        @SerializedName("address")
+        @Expose
+        private String address;
+        @SerializedName("chainId")
```
Comment (seabornlee) [inline app/src/main/java/com/alphawallet/app/entity/TokensMapping.java:15]: A blank line should be here.

File: app/src/main/java/com/alphawallet/app/entity/TokensMapping.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.List;
+
+public class TokensMapping {
+
+    public class Contract {
+
+        @SerializedName("address")
+        @Expose
+        private String address;
```
Comment (seabornlee) [inline app/src/main/java/com/alphawallet/app/entity/TokensMapping.java:14]: A blank line should be here.

File: app/src/main/java/com/alphawallet/app/entity/TokensMapping.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+public class TokensMapping {
+
+    public class Contract {
+
+        @SerializedName("address")
```
Comment (seabornlee) [inline app/src/main/java/com/alphawallet/app/entity/TokensMapping.java:12]: Can we remove @SerializedName when the names between JSON and Java object are identical?
https://www.javadoc.io/doc/com.google.code.gson/gson/2.6.2/com/google/gson/annotations/SerializedName.html

File: app/src/main/java/com/alphawallet/app/entity/TokensMapping.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        @Expose
```
Comment (seabornlee) [inline app/src/main/java/com/alphawallet/app/entity/TokensMapping.java:13]: @Expose is also not required.
image

File: app/src/main/java/com/alphawallet/app/entity/TokensMapping.java (hunk 3 above)
Comment (JamesSmartCell) [reply app/src/main/java/com/alphawallet/app/entity/TokensMapping.java:12]: @seabornlee can you add a refactor PR with these changes? I will push this PR as-is

Comment (seabornlee) [reply app/src/main/java/com/alphawallet/app/entity/TokensMapping.java:12]: Sure.

**Review by JamesSmartCell — APPROVED** [review]

**Pull request merged**